app/models/framework_response.rb

```diff
@@ -13,27 +16,17 @@ class FrameworkResponse < VersionedModel
 
   belongs_to :parent, class_name: 'FrameworkResponse', optional: true
   has_and_belongs_to_many :framework_flags, autosave: true
-  validates_each :value, on: :update do |record, _attr, value|
-    record.errors.add(:value, :blank) if requires_value?(value, record)
-  end
 
   after_validation :set_responded_value, on: :update
 
-  def self.requires_value?(value, record)
-    return false if value.present? || !record.framework_question.required
-
-    return true if record.parent.blank?
-
-    record.parent.option_selected?(record.framework_question.dependent_value)
-  end
-
   def update_with_flags!(value)
     ActiveRecord::Base.transaction do
-      old_value = self.value
       self.value = value
+      return unless value_text_changed? || value_json_changed?
```

Comment:
so I double checked why I needed the `old_value = self.value` instead of `value_text_changed? || value_json_changed?` and it seems that this is required for jsonb responses, as it always picks this up as changed although it hasn't so I think we will need to revert to the old implementation of using `old_value`